src/Impostor.Server/Events/Game/Player/PlayerMovementEvent.cs

```diff
@@ -1,24 +1,71 @@
-﻿using Impostor.Api.Events.Player;
+using System;
+using System.Collections.Concurrent;
+
+using Impostor.Api.Events.Player;
 using Impostor.Api.Games;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Inner.Objects;
+using Impostor.Hazel;
 
 namespace Impostor.Server.Events.Player
 {
     // TODO: Finish and use event, needs to be pooled
-    public class PlayerMovementEvent : IPlayerEvent
+    public class PlayerMovementEvent : IPlayerMovementEvent , IRecyclable , IDisposable
     {
+        private static readonly ConcurrentBag<PlayerMovementEvent> items = new ConcurrentBag<PlayerMovementEvent>();
+        private static int counter = 0;
```

Comment:
Should `counter` be static here? Or local to the classes instance, as I'd assume `X` amount of movements per player would be much more useful, instead of `X` amount of movements in the game?